### PR TITLE
lib: edge_impulse: Expand ei_wrapper API

### PR DESCRIPTION
**Pull request by MarekPieta**
Expand API to provide input data sampling frequency, all labels and result data associated with every label. Change also updates related application, sample and test. Documentation is updated too.

Jira: NCSDK-12108

### CODE REVIEW

**Review by pdunaj — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -303,6 +303,10 @@ Other libraries
 
   * Increased number of supported Event Manager events.
 
+* :ref:`ei_wrapper` library:
+
+  * Expand API to provide: input data sampling frequency, every label used by the machine learning model and result data associated with every label.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:308]:
```suggestion
  * Expanded API to provide information about input data sampling frequency, every label used by the machine learning model, and results associated with every label.
```

File: doc/nrf/ug_edge_impulse.rst [hunk 2]
```diff
@@ -69,8 +69,10 @@ Complete the following steps to configure the building process:
 #. Enable and specify the Uniform Resource Identifier (URI) in the :kconfig:`CONFIG_EDGE_IMPULSE_URI` Kconfig option.
    You can set it to one of the following values:
 
-   * An absolute path to a file in the local file system.
+   * An absolute or relative path to a file in the local file system.
      For this variant, you must download the :file:`zip` file manually and place it under path defined by the Kconfig option.
+     The relative path starts from application source directory (``APPLICATION_SOURCE_DIR``).
```
Comment (greg-fer) [inline doc/nrf/ug_edge_impulse.rst:74]:
```suggestion
     The relative path is tracked from the application source directory (``APPLICATION_SOURCE_DIR``).
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: lib/edge_impulse/ei_wrapper.cpp [hunk 3]
```diff
@@ -312,50 +336,133 @@ static bool can_read_result(void)
...
+	if (first_classification_get) {
+		first_classification_get = false;
+		cur_idx = get_result_with_highest_value(NULL);
+		__ASSERT_NO_MSG(cur_idx < RESULT_LABEL_COUNT);
+		found = true;
+	} else {
+		/* If there is label with higher index and the same value, return it. */
+		float cur_val = ei_result.classification[cur_idx].value;
+
+		for (size_t idx = cur_idx + 1; idx < RESULT_LABEL_COUNT; idx++) {
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply lib/edge_impulse/ei_wrapper.cpp:388]: Thanks for the suggestions. This seems simpler now :)

File: lib/edge_impulse/ei_wrapper.cpp [hunk 4]
```diff
@@ -312,50 +337,112 @@ static bool can_read_result(void)
...
+			return idx;
+		}
+
+		if ((val < limit) && (val > max_val)) {
+			max_idx = idx;
+			max_val = val;
+			found = true;
+		}
+
+		if ((max_val == -INFINITY) && (val == -INFINITY) && (!found)) {
```
Comment (pdunaj) [inline lib/edge_impulse/ei_wrapper.cpp:369]: discussed offline - let's combine two cases

**Review by pdunaj — CHANGES_REQUESTED** [review]

File: lib/edge_impulse/ei_wrapper.cpp (hunk 4 above)
Comment (MarekPieta) [reply lib/edge_impulse/ei_wrapper.cpp]: Fixed

**Review by pdunaj — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Pull request merged**